extensions/guacamole-auth-sso/modules/guacamole-auth-sso-cas/src/main/java/org/apache/guacamole/auth/cas/CASResource.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.guacamole.auth.cas;
+
+import com.google.inject.Inject;
+import javax.ws.rs.core.Response;
+import org.apache.guacamole.GuacamoleException;
+import org.apache.guacamole.auth.sso.SSOResource;
+
+/**
+ * REST API resource that automatically redirects users to the CAS login
+ * endpoint.
+ */
+public class CASResource implements SSOResource {
+
+    /**
+     * Service for authenticating users using CAS.
+     */
+    @Inject
+    private AuthenticationProviderService authService;
+
+    @Override
+    public Response redirectToIdentityProvider() throws GuacamoleException {
+        return Response.seeOther(authService.getLoginURI()).build();
```

Comment:
Isn't this exactly the same code as `OpenIDResource `? Maybe it could be abstracted out to an abstract base class or something?

Reply:
Good catch - I just tried this and it is definitely an improvement.